### PR TITLE
response_writer: support http/2 push

### PR DESCRIPTION
**Pull request by jeffprestes**
PR to response the https://github.com/go-macaron/macaron/issues/136 feature request

### CODE REVIEW

**Comment by unknwon** [comment]
Action items after merge:
- [ ] Update `.travis.yml` to remove Go 1.6, 1.7 and add Go 1.13, 1.14.
- [ ] Change README and docs to state minimum support version is Go 1.8.

File: response_writer.go [hunk 1]
```diff
@@ -112,3 +113,12 @@ func (rw *responseWriter) Flush() {
 		flusher.Flush()
 	}
 }
+
+func (rw *responseWriter) Push(target string, opts *http.PushOptions) error {
+	pusher, ok := rw.ResponseWriter.(http.Pusher)
+	if !ok {
+		return fmt.Errorf("the ResponseWriter doesn't support the Pusher interface")
```
Comment (unknwon) [inline response_writer.go]: The use of `fmt.Errorf` seems not useful, would you mind change this line and line 95 to both use `errors.New` instead?

File: response_writer.go [hunk 2]
```diff
@@ -112,3 +113,12 @@ func (rw *responseWriter) Flush() {
...
+	}
+	pusher.Push(target, opts)
+	return nil
```
Comment (unknwon) [inline response_writer.go]:
```suggestion
	return pusher.Push(target, opts)
```

File: response_writer_test.go [hunk 3]
```diff
@@ -185,4 +185,11 @@ func Test_ResponseWriter(t *testing.T) {
 		So(resp.Code, ShouldEqual, http.StatusOK)
 		So(resp.Body.String(), ShouldEqual, "data: Hello\n\ndata: Hello\n\n")
 	})
+
+	Convey("Response writer with http/2 push", t, func() {
+		resp := httptest.NewRecorder()
+		rw := NewResponseWriter("GET", resp)
+		_, ok := rw.(http.Pusher)
+		So(ok, ShouldBeTrue)
```
Comment (unknwon) [inline response_writer_test.go]: No test case for the call of `pusher.Push` method?

File: response_writer.go (hunk 1 above)
Comment (jeffprestes) [reply response_writer.go]: Done.

File: response_writer_test.go (hunk 3 above)
Comment (jeffprestes) [reply response_writer_test.go]: I will not lie to you. As I never used Convoy I got a little bit confused how to use it in this more complex use case. As I have other demands from my employer if you could help me out on this or give a waiver in this method I would appreciate it

Comment (unknwon) [reply response_writer_test.go]: No worries! I'll see what I can do and push to this PR.

**Comment by unknwon** [comment]
Travis CI is broken, I'm going to merge as-is, and might fix test in the future.

**Pull request merged**